src/main/java/com/cityelf/service/WaterForecastService.java

```diff
@@ -0,0 +1,91 @@
+package com.cityelf.service;
+
+import com.cityelf.exceptions.ForecastAlreadyExistsException;
+import com.cityelf.exceptions.ForecastNotFoundException;
+import com.cityelf.model.Address;
+import com.cityelf.model.WaterForecast;
+import com.cityelf.repository.WaterForecastRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class WaterForecastService {
+
+  @Autowired
+  private WaterForecastRepository waterForecastRepository;
+
+  public WaterForecastService() {
```

Comment:
No need for default constructor if there is no other one. It will be added by the compiler automatically.